h5netcdf/tests/test_h5netcdf.py

```diff
@@ -108,7 +108,7 @@ def array_equal(a, b):
 def is_h5py_char_working(tmp_netcdf, name):
     h5 = get_hdf5_module(tmp_netcdf)
     # https://github.com/Unidata/netcdf-c/issues/298
-    with h5.File(tmp_netcdf, "r") as ds:
+    with h5.File(tmp_netcdf) as ds:
```

Comment:
This should probably still explicitly set `mode='r'`, unless you are writing a test explicitly to check for the warning message -- which I'm pretty sure is not the case here.

Reply:
Sure, added. I've added an explicit test for the FutureWarning too.